x/dex/spec/README.md

```diff
@@ -1,21 +1,46 @@
 ## Abstract
-
-TODO: Populate dex README Contents below.
-
-## Contents
+`dex` module is responsible for matching orders for registered contracts.
 
 ## Concepts
-
-## State
-
-## Messages
-
-## Events
+### Frequent Batch Auction
+The traditional implementation of exchange logic would look something like the following:
+1. User A sends an order placement transaction
+2. User B sends another order placement transaction
+3. User A's transaction is processed by matching it against the order book state and settle accordingly
+4. User B's transaction is processed similarly
+
+Sei's `dex` module takes a different approach:
+1. User A sends an order placement transaction
+2. User B sends another order placement transaction
+3. User A's transaction is processed by simply adding the order to an in-memory queue, without matching
+4. User B's transaction is processed similarly
+5. At the end of the block that contains both transactions, the in-memory queue as a whole will be matched against the order book state
+
+### Contract Registration
+Since `dex` only provides order matching logic, product logic specific to individual protocols still needs to be defined in CosmWasm contracts. As such, `dex` offers a way to inform the protocol contracts about order placement and matching results. `dex` achieves this by requiring contracts that want to leverage `dex`'s order matching logic to explicitly register via a special transaction type `MsgRegisterContract`.
```

Comment:
can we elaborate what specifically can be registered here? i.e. tick size, quantity size and dependencies

Reply:
and where those variables are registered/accessed + how they are used